scheduler/scheduler.cpp

```diff
@@ -1766,16 +1770,19 @@ static int prepare_broadcast_reply(char* buf, const char* netname)
     }
 }
 
-static void broadcast_scheduler_version()
+static void broadcast_scheduler_version(const char* netname)
 {
-    const int schedbuflen = 4 + sizeof(uint64_t);
+    char length_netname = sizeof(netname);
+    const int schedbuflen = 5 + sizeof(uint64_t) + length_netname;
     char buf[ schedbuflen ];
     buf[0] = 'I';
     buf[1] = 'C';
     buf[2] = 'E';
     buf[3] = PROTOCOL_VERSION;
     uint64_t tmp_time = starttime;
     memcpy(buf + 4, &tmp_time, sizeof(uint64_t));
+    buf[4 + sizeof(uint64_t)] = length_netname;
+    memcpy(buf + 5 + sizeof(uint64_t), netname, length_netname);
     DiscoverSched::broadcastData(scheduler_port, buf, sizeof(buf));
```

Comment:
When we send a "version 36+" broadcast with the netname, I believe that the old scheduler is still able to use that broadcast data and it will just ignore the additional data that it doesn't use. The call to `recvfrom(... buf, buflen, ...) ` will fill the buffer with only the data that it's expecting (provided by buflen) and any additional data received will be discarded(won't be available for a future call to recvfrom()).